src/dbnode/client/session_proto_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package client
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3/src/x/ident"
+	"github.com/m3db/m3/src/dbnode/testdata/prototest"
+	"github.com/m3db/m3/src/dbnode/encoding"
+	xtime "github.com/m3db/m3/src/x/time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/m3db/m3/src/dbnode/ts"
+	"github.com/m3db/m3/src/dbnode/generated/thrift/rpc"
+	"github.com/m3db/m3/src/dbnode/storage/namespace"
+)
+
+var (
+	testNamespace     = ident.StringID("session_prototest_ns")
+	// Relative path to $pwd to reach testdata/prototest, $pwd is current directory.
+	testSchemaHistory = prototest.NewSchemaHistory("../")
```

Comment:
Hm, if it's relative shouldn't it be `prototest.NewSchemaHistory("../testdata/prototest")`?  

This is a little bit of a strange contract for the `prototest.NewSchemaHistory(...)` function.

Could you not perhaps just inline the protobuf messages in prototest package rather than have it read a file?  Then it would avoid all this strange path stuff.

You could do in the `prototest` package just some declarations like so:
```go
const testProto = `
message My Proto {
  ...
}
`
```

Reply:
the protoparse package we use do not take string.

Reply:
Hm interesting, how are we pushing schema updates through etcd out of interest if we can only load it from the filesystem?

Reply:
(Unfortunate that there's no API for non-file based schemas https://godoc.org/github.com/jhump/protoreflect/desc/protoparse)

Reply:
writing to a temp dir/file seem to be a good solution. for test, we will use it.